src/main/scala/org/allenai/pipeline/Producer.scala

```diff
@@ -47,6 +47,36 @@ trait Producer[T] extends PipelineStep with CachingEnabled with Logging {
     returnValue
   }
 
+  /**
+   * Lift a function to producers. This gives you ability to fully customize the step info
+   */
+  def map[R](stepInfo: PipelineStepInfo, f: T => R): Producer[R] = {
```

Comment:
In particular, StepInfo.dependencies should be the original Producer you started with.  And we need to make sure we're playing nice with the .persist() method (we might be already, but unclear)